src/Viewer/services/decoder/FileManager.js

```diff
@@ -176,6 +190,37 @@ class FileManager {
         this.state.numberOfEvents = this._logEventOffsets.length;
     };
 
+    /**
+     * Given a ref timestamp, find the first log event whose timestamp is 
+     * greater or equal to the ref timestamp.
+     * Since all the log events are naturally sorted w.r.t. the timestamp,
+     * we can use binary search.
+     * Return the index of the log event.
+     * If the log event doesn't exist, the function should return null
+     */
+    _findFirstLogEventWithTimestamp (refTimestamp) {
```

Comment:
Can we add a few unit tests (don't need to go overboard) for this method? You can see an example of how we write unit tests in `FourByteVarBuf.test.js`